python/ray/serve/_private/logging_utils.py

```diff
@@ -74,12 +75,20 @@ def format(self, record: logging.LogRecord) -> str:
             ]
         if SERVE_LOG_ROUTE in record.__dict__:
             record_format[SERVE_LOG_ROUTE] = SERVE_LOG_RECORD_FORMAT[SERVE_LOG_ROUTE]
+
         if SERVE_LOG_APPLICATION in record.__dict__:
             record_format[SERVE_LOG_APPLICATION] = SERVE_LOG_RECORD_FORMAT[
                 SERVE_LOG_APPLICATION
             ]
 
-        record_format[SERVE_LOG_MESSAGE] = SERVE_LOG_RECORD_FORMAT[SERVE_LOG_MESSAGE]
+        if SERVE_LOG_EXTRA_FIELDS in record.__dict__:
+            for k, v in record.__dict__[SERVE_LOG_EXTRA_FIELDS].items():
+                record_format[k] = v
+
+        if SERVE_LOG_MESSAGE in record.__dict__:
+            record_format[SERVE_LOG_MESSAGE] = (
+                SERVE_LOG_RECORD_FORMAT[SERVE_LOG_MESSAGE] % record.__dict__
+            )
```

Comment:
Why is this changed? I thought the entries of `SERVE_LOG_RECORD_FORMAT` are not "filled in" with values yet

Reply:
this is to early-fill in the values and jsonize the value to avoid non-jsonizable value in the log content.